Tests/SourceKitLSPTests/ExecuteCommandTests.swift

```diff
@@ -141,120 +159,362 @@ final class ExecuteCommandTests: XCTestCase {
+        command.arguments?.append(metadata.encodeToLSPAny())
 
-      let result = try await project.testClient.send(request)
+        let request = ExecuteCommandRequest(command: command.command, arguments: command.arguments)
 
-      guard let resultArray: [RefactoringEdit] = Array(fromLSPArray: result ?? .null) else {
-        XCTFail(
-          "Result is not an array. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
-        )
-        return
-      }
+        if peekDocuments {
+          let expectation = self.expectation(description: "Handle Peek Documents Request")
 
-      XCTAssertEqual(
-        resultArray.count,
-        1,
-        "resultArray is empty. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
-      )
-      XCTAssertEqual(
-        resultArray.only?.newText,
-        "(1 + 2, \"1 + 2\")",
-        "Wrong macro expansion. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
-      )
+          let peekDocumentsRequestURIs = ThreadSafeBox<[DocumentURI]?>(initialValue: nil)
 
-      try await fulfillmentOfOrThrow([expectation])
+          project.testClient.handleSingleRequest { (req: PeekDocumentsRequest) in
+            peekDocumentsRequestURIs.value = req.locations
+            expectation.fulfill()
+            return PeekDocumentsResponse(success: true)
+          }
 
-      let url = try XCTUnwrap(
-        showDocumentRequestURI.value?.fileURL,
-        "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
-      )
+          _ = try await project.testClient.send(request)
+
+          try await fulfillmentOfOrThrow([expectation])
+
+          let urls = try XCTUnwrap(
+            peekDocumentsRequestURIs.value?.map {
+              return try XCTUnwrap(
+                $0.fileURL,
+                "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+              )
+            },
+            "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
+
+          let filesContents = try urls.map { try String(contentsOf: $0, encoding: .utf8) }
+
+          XCTAssertEqual(
+            filesContents,
+            [
+              "@DictionaryStorageProperty",
+              "@DictionaryStorageProperty",
+              "var _storage: [String: Any] = [:]",
+            ],
+            "Files doesn't contain correct macro expansion. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
+
+          XCTAssertEqual(
+            urls.map { $0.lastPathComponent }.sorted(),
+            [
+              "MyMacroClient_L11C3-L11C3.swift",
+              "MyMacroClient_L12C3-L12C3.swift",
+              "MyMacroClient_L13C1-L13C1.swift",
+            ].sorted(),
+            "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
```

Comment:
Why do we need to sort the URLs but not the file contents?

Reply:
With file contents, we have to ensure that the macro expands in the same order as we expect. I don't think we should sort them, since it won't allows us to test the order of expansion.

However, with the file names, whatever the order it maybe, as long as the line and character numbers are correct, we are good to go. So I `sorted` them to let the equality check to pass, as long as they contain the same elements, irrespective of the order.

Let me know if I overlooked anything here.

Reply:
But the order of the file names is always deterministic, right? So it would be fine to compare them without sorting, right? That way we would effectively also be checking that the correct contents are in the correct file.

Reply:
Makes sense 👍🏻